lib/active_fedora/file.rb

```diff
@@ -63,7 +63,15 @@ def ldp_connection
     # If this file has a parent with ldp#contains, we know it is not new.
     # By tracking exists we prevent an unnecessary HEAD request.
     def new_record?
-      !@exists && ldp_source.new?
+      !@exists && ldp_new_record?
```

Comment:
@jcoyne this is what I needed to do to get the delete for direct_container to work. The call to `target.delete(record)` in `collection_association.rb` internally calls `new_record?` **after** the actual record has been deleted. I believe when Ruby processes target.delete(record) it's using the `==` operator defined in file.rb which in turns calls new_record.